lib/checkstring.cpp

```diff
@@ -309,7 +315,8 @@ void CheckString::checkIncorrectStringCompare()
                 incorrectStringBooleanError(tok->next(), tok->strAt(1));
             } else if (Token::Match(tok, "if|while ( %str%|%char% )") && !tok->tokAt(2)->getValue(0)) {
                 incorrectStringBooleanError(tok->tokAt(2), tok->strAt(2));
-            }
+            } else if (tok->str() == "?" && tok->astOperand1() && Token::Match(tok->astOperand1(), "%str%|%char%"))
```

Comment:
It is redundant to check if `tok->astOperand1()` is null before the call to `Token::Match`.

Reply:
Thanks, fixed. Seems `CheckInternal::checkRedundantTokCheck()` only detects this for simple patterns like `tok && Token::Match(tok, "bla2")`, and not more complex patterns like `a && tok && Token::Match(tok, "bla2")` (since in that case, `astOperand1()` of the second `&&` is the first `&&`).